sklego/model_selection.py

```diff
@@ -248,8 +248,17 @@ def get_split_info(X, indices, j, part, summary):
         return pd.DataFrame(summary)
 
 
-class KlusterFoldValidation:
-    """KlusterFold cross validator. Create folds based on provided cluster method
+def KlusterFoldValidation(**kwargs):
+    warn(
+        "Please use `ClusterFoldValidation` instead of `KlusterFoldValidation`."
+        "We will use correct spelling going forward and `KlusterFoldValidation` will be deprecated.",
+        DeprecationWarning,
+    )
+    return ClusterFoldValidation(**kwargs)
+
+
+class ClusterFoldValidation:
+    """Cross validator that create folds based on provided cluster method.
```

Comment:
Couple of addition I can think of are:
- Adding a future release version 
- Extend the description a tiny bit

```suggestion
class ClusterFoldValidation:
    """Cross validator that creates folds based on provided cluster method. Each fold train/test indices are generated 
    by the clustering algorithm. This ensures that data points in the same cluster are not split across different folds.
    
    !!! info "New in version 0.9.0"
```